### PR TITLE
feat(aws): Adding support for launch templates in UpdateSecurityGroup…

### PR DESCRIPTION
**Pull request by jeyrschabu**
…sForServerGroupStage

- added UpdateLaunchTemplateTask to handle security groups for launch template backed server groups

### CODE REVIEW

**Comment by jeyrschabu** [comment]
Ref: https://github.com/spinnaker/clouddriver/pull/4710

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/UpdateLaunchConfigTask.groovy [hunk 1]
```diff
@@ -21,6 +21,7 @@ import com.netflix.spinnaker.orca.api.pipeline.Task
 import com.netflix.spinnaker.orca.api.pipeline.models.StageExecution
```
Comment (cfieber) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/UpdateLaunchConfigTask.groovy:21]: I'd be tempted to rename this class to something like AbstractSomethingSomethingLaunchTask and then have two concrete subclasses.

Feels like some slight risk that someone could accidentally break launch template by doing something for launch config

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/UpdateLaunchTemplateTask.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    ctx.put("kato.last.task.id", taskId);
+    ctx.put(
+        "deploy.server.groups",
+        Collections.singletonMap(region, Collections.singleton(serverGroupName)));
+    return ctx;
+  }
+
+  @Override
+  public String getOperation() {
+    return "updateLaunchTemplate";
```
Comment (cfieber) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/UpdateLaunchTemplateTask.java:53]:
```suggestion
    return OPERATION;
```

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/pipeline/UpdateSecurityGroupsForServerGroupStage.java [hunk 3]
```diff
@@ -45,9 +46,15 @@
...
   public void taskGraph(@Nonnull StageExecution stage, @Nonnull TaskNode.Builder builder) {
-    builder
-        .withTask("updateLaunchConfig", UpdateLaunchConfigTask.class)
-        .withTask("updateInstances", UpdateInstancesTask.class);
+    Boolean hasLaunchTemplate =
+        (Boolean) stage.getContext().getOrDefault("hasLaunchTemplate", false);
+    if (hasLaunchTemplate) {
+      builder.withTask("updateLaunchTemplate", UpdateLaunchTemplateTask.class);
+    } else {
+      builder.withTask("updateLaunchConfig", UpdateLaunchConfigTask.class);
```
Comment (cfieber) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/pipeline/UpdateSecurityGroupsForServerGroupStage.java:54]: could probably use the OPERATION constant here too

**Review by cfieber — APPROVED** [review]

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/UpdateLaunchConfigTask.groovy (hunk 1 above)
Comment (jeyrschabu) [reply orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/UpdateLaunchConfigTask.groovy:21]: That's a good point.

**Pull request merged**